src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveIRPass.cs

```diff
@@ -94,18 +89,8 @@ public override void VisitDirective(DirectiveIRNode node)
 
                     node.Parent.Children.Insert(sectionIndex, defineSectionStartStatement);
                 }
-                else if (string.Equals(node.Name, CSharpCodeParser.AddTagHelperDirectiveDescriptor.Name, StringComparison.Ordinal))
-                {
-                    node.Parent.Children.Remove(node);
-                }
-                else if (string.Equals(node.Name, CSharpCodeParser.RemoveTagHelperDirectiveDescriptor.Name, StringComparison.Ordinal))
-                {
-                    node.Parent.Children.Remove(node);
-                }
-                else if (string.Equals(node.Name, CSharpCodeParser.TagHelperPrefixDirectiveDescriptor.Name, StringComparison.Ordinal))
-                {
-                    node.Parent.Children.Remove(node);
-                }
+
+                node.Parent.Children.Remove(node);
```

Comment:
Hmm, this kinda overloads this passes purpose to also clean up `DirectiveIRNodes`. This could be fine for now but we may want to move it to a different pass later; that way users can get our default directive support without us nuking every `DirectiveIRNode` under the sun 😄